File: retroauth/src/test/java/com/andretietz/retroauth/AuthenticationHandlerTest.java
package com.andretietz.retroauth;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class AuthenticationHandlerTest {

    @Mock
    MethodCache<String> methodCache;
    @Mock
    OwnerManager<String, String> ownerManager;
    @Mock
    TokenStorage<String, String, String> tokenStorage;
    @Mock
    TokenProvider<String, String, String> provider;
    @Mock
    TokenTypeFactory<String> typeFactory;

    @Test
    public void allocate() {
        AuthenticationHandler<String, String, String> authHandler =
                new AuthenticationHandler<>(methodCache, ownerManager, tokenStorage, provider, typeFactory);
        Assert.assertNotNull(authHandler.getMethodCache());
        Assert.assertNotNull(authHandler.getOwnerManager());
        Assert.assertNotNull(authHandler.getTokenStorage());
        Assert.assertNotNull(authHandler.getProvider());
        Assert.assertNotNull(authHandler.getTypeFactory());
    }
}

File: retroauth/src/test/java/com/andretietz/retroauth/AuthenticationHandlerTest.kt
package com.andretietz.retroauth

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class AuthenticationHandlerTest {

    @Mock
    private lateinit var methodCache: MethodCache<String>
    @Mock
    private lateinit var ownerManager: OwnerManager<String, String>
    @Mock
    private lateinit var tokenStorage: TokenStorage<String, String, String>
    @Mock
    private lateinit var provider: TokenProvider<String, String, String>

    @Test
    fun allocate() {
        val authHandler = AuthenticationHandler(methodCache, ownerManager, tokenStorage, provider)
        Assert.assertNotNull(authHandler.methodCache)
        Assert.assertNotNull(authHandler.ownerManager)
        Assert.assertNotNull(authHandler.tokenStorage)
        Assert.assertNotNull(authHandler.provider)
    }
}
